src/core/simulation.jl

```diff
@@ -452,8 +449,10 @@ function _check_steps(
     for (stage_number, stage_name) in sim.sequence.order
         stage = sim.stages[stage_name]
         execution_counts = get_executions(stage)
-        @assert length(findall(x -> x == stage_number, sim.sequence.execution_order)) ==
-                execution_counts
+        transitions =
+            sim.sequence.execution_order[vcat(1, diff(sim.sequence.execution_order)) .== 1]
```

Comment:
@claytonpbarrows what's the purpose of this logic?

Reply:
1. find all the times that you execute a subsequent stage ( find 1->2 transitions, not 2->1)
 2. the total times that a stage is executed in the entire simulation / the number of simulation steps should be equal to the expected number of stage executions per step

The previous logic was failing when we ran a simulation that ran multiple executions of some stages.

Reply:
We need to add more tests for all potential sequences to make sure that everything works as expected.